client/src/components/admin/tables/ProjectRow.tsx

```diff
@@ -67,19 +74,32 @@ const ProjectRow = ({ project, idx, checked, handleCheckedChange }: ProjectRowPr
         }
     };
 
+    useEffect(() => {
+        // Filter flags to find all matching ones
+        const matchingFlags = flags.filter((flag) => flag.project_id === project.id);
+        // Set matching flag count
+        const matchingFlagCount = matchingFlags.length;
+        // If there is at least one matching flag, set the first one as the matching flag
+        if (matchingFlagCount > 0) {
+            setMatchingFlag(matchingFlags[0]); // Update matchingFlag state to the first match
+            setMatchingFlagCount(matchingFlagCount); // Update matchingFlagCount state
+            setProjectIDState(matchingFlags[0].project_id); // Update projectIDState based on first matching flag
+        } else {
+            setMatchingFlag(undefined); // No match found
+            setProjectIDState(''); // Clear project ID state
+        }
+    }, [flags, project]); // Recalculate when flags or project changes
```

Comment:
also this is entirely too many comments; I don't need to know what happens on EVERY SINGLE LINE. See what I did in [`JudgesTable.tsx`](https://github.com/hackutd/jury/blob/53c28b6c4d7291ef3f3e8fa3f1210506ee240d3d/client/src/components/admin/tables/JudgesTable.tsx#L24) for the `updateSort` function. If you have reasonable variable names and code flow, it should be readable enough without comments on EACH line.